systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeST.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.upgrade;
+
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.Environment;
+import io.strimzi.systemtest.enums.OlmInstallationStrategy;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaBasicExampleClients;
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaBridgeExampleClients;
+import io.strimzi.systemtest.resources.operator.OlmResource;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.FileUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.StatefulSetUtils;
+import io.strimzi.systemtest.utils.specific.OlmUtils;
+import io.strimzi.test.k8s.KubeClusterResource;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.json.JsonObject;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.strimzi.systemtest.resources.ResourceManager.kubeClient;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+public class OlmUpgradeST extends AbstractUpgradeST {
+
+    private static final Logger LOGGER = LogManager.getLogger(OlmUpgradeST.class);
+
+    private final String namespace = "olm-upgrade-namespace";
+    private final String producerName = "producer";
+    private final String consumerName = "consumer";
+    private final String topicUpgradeName = "topic-upgrade";
+    private final int messageUpgradeCount =  50_000; // 10k ~= 23s, 50k ~= 115s
+    private final KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBridgeExampleClients.Builder()
+        .withProducerName(producerName)
+        .withConsumerName(consumerName)
+        .withBootstrapAddress(KafkaResources.plainBootstrapAddress(CLUSTER_NAME))
+        .withTopicName(topicUpgradeName)
+        .withMessageCount(messageUpgradeCount)
+        .withDelayMs(1)
+        .build();
+
+    @ParameterizedTest(name = "testUpgradeStrimziVersion-{0}-{1}")
+    @MethodSource("loadJsonUpgradeData")
+    void testChainUpgrade(String fromVersion, String toVersion, JsonObject testParameters) {
+
+        int clusterOperatorVersion = Integer.parseInt(fromVersion.split("\\.")[1]);
+        // only 0.|18|.0 and more is supported
+        assumeTrue(clusterOperatorVersion >= 18);
```

Comment:
What about Strimzi 1.0.0 ?

Reply:
changed to `assumeTrue(clusterOperatorMajorVersion >= 0 && clusterOperatorMiddleVersion >= 18);`